quill-cassandra/src/main/scala/io/getquill/source/cassandra/Decoders.scala

```diff
@@ -0,0 +1,57 @@
+package io.getquill.source.cassandra
+
+import io.getquill.source.sql.SqlSource
+import com.datastax.driver.core.Row
+import com.datastax.driver.core.BoundStatement
+
+trait Decoders {
+  this: SqlSource[_, _, Row, BoundStatement] =>
+
+  private def decoder[T](f: Row => Int => T): Decoder[T] =
+    new Decoder[T] {
+      def apply(index: Int, row: Row) =
+        f(row)(index)
+    }
+
+  implicit def optionDecoder[T](implicit d: Decoder[T]): Decoder[Option[T]] =
+    new Decoder[Option[T]] {
+      def apply(index: Int, row: Row) = {
+        row.isNull(index) match {
+          case true  => None
+          case false => Some(d(index, row))
+        }
+      }
+    }
+
+  implicit val stringDecoder = decoder(_.getString)
+  implicit val bigDecimalDecoder: Decoder[BigDecimal] =
+    new Decoder[BigDecimal] {
+      def apply(index: Int, row: Row) =
+        row.getDecimal(index)
+    }
+  implicit val booleanDecoder = decoder(_.getBool)
+  implicit val byteDecoder: Decoder[Byte] =
+    new Decoder[Byte] {
+      def apply(index: Int, row: Row) =
+        row.getBytes(index).get
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/fwbrasil/quill/file/1706967300/issues/source?bid=2746728&fileBranchId=3046581#l36)